Bump to 1.11.0

gkurz opened the pull request:
Provide a script to bump the OSC version in the whole repo.

Bump to 1.11.0.

gkurz committed:
Release 1.11.0: update the version everywhere

Run :

    ./scripts/bump-osc-version.sh 1.11.0 1.10.2

This is preparatory work for the upcoming release.

Signed-off-by: Greg Kurz <groug@kaod.org>

gkurz committed:
Improve OSC version bump experience

Add some more ## OSC_VERSION annotations and provide a script that bumps
the version everywhere in the source tree.

Signed-off-by: Greg Kurz <groug@kaod.org>

littlejawa commented on Dockerfile:
```diff
@@ -41,7 +41,7 @@ RUN getent group nonroot || groupadd -o -g 499 nonroot
 LABEL name="openshift-sandboxed-containers/osc-rhel9-operator" \
 cpe="cpe:/a:redhat:confidential_compute_attestation:1.10::el9" \
-version="1.10.1" \
+version="1.11.0" \
```
I'm thinking we may want to keep the container LABEL version to "1.11", without the "z" version, so that an image doesn't seem tied to a specific "z" version.
This allows us to re-release the same images in a subsequent release, if only a subset of images needed to be modified, without making people confused, thinking they're not using the right one.

This is nitpicking honestly, because it really seems that this "version" label is not used anywhere :-/

gkurz replied:
> I'm thinking we may want to keep the container LABEL version to "1.11", without the "z" version, so that an image doesn't seem tied to a specific "z" version. 

But an image can also be tied to a specific "z" version in case of bug fix, can't it ?

>This allows us to re-release the same images in a subsequent release, if only a subset of images needed to be modified, without making people confused, thinking they're not using the right one.

If the goal is to avoid to release an image that has same payload, would it be possible to re-label the existing image in this case ?

> This is nitpicking honestly, because it really seems that this "version" label is not used anywhere :-/

The "version" label is _just_ a user visible piece of information. It seems to be frequently present in images, e.g. ubi9.

dbkreling replied:
I understand and agree in **not** having discrepant versions available to customers with different "z" digit when they refer to the same codebase. But I want to confirm the customers will be able to differentiate their current installation version.

Images contain 3 different tags visible in [the customer portal](https://catalog.redhat.com/en/software/containers/openshift-sandboxed-containers/osc-rhel9-operator/63eb68805620225430ca2c97?container-tabs=gti&gti-tabs=registry-tokens) (_latest_, _version-with-timestamp_ and _git_short_sha_). If the customers are able to compare their current installation with the latest released image available, I think we definitely should go with the two digit (X.Y) pattern for the version. If not, we should think how we could differentiate them.

littlejawa replied:
The LABEL we see here is attached to the image. You can see it when inspecting the container image itself. I don't think we can re-label the image without re-building it.
It is different than a tag that we can apply/modify on an existing image.
We will be tagging images with x.y.z when we release them, even if they were already released before, but the LABEL could stay the same.

gkurz replied:
Ok. I'll convert the version labels to be x.y only in a preparatory commit in the same PR.

dbkreling commented on must-gather/Dockerfile:
```diff
@@ -24,7 +24,7 @@ COPY node-gather/node-gather-ds.yaml /etc/
 # Red Hat labels
 LABEL name="openshift-sandboxed-containers/osc-must-gather-rhel9" \
 cpe="cpe:/a:redhat:confidential_compute_attestation:1.10::el9" \
```
Should we bump 1.10 to 1.11 in this line?

littlejawa replied:
Eventually, yes.
But the cpe is defined by our ReleasePlanAdmission version, and this gets updated separately (and potentially closer to the release).
I think this should get updated along with the ReleasePlanAdmission, not with the rest of our version bumps.

littlejawa replied:
For reference: 
cpe definition: (based on ReleasePlanAdmission version)
https://gitlab.cee.redhat.com/releng/konflux-release-data/-/blob/main/prodsec/ose-osc.yaml#L3

Prod ReleasePlanAdmission version:
https://gitlab.cee.redhat.com/releng/konflux-release-data/-/blob/main/config/stone-prd-rh01.pg1f.p1/product/ReleasePlanAdmission/ose-osc/osc-prod.yaml#L19

Stage ReleasePlanAdmission version:
https://gitlab.cee.redhat.com/releng/konflux-release-data/-/blob/main/config/stone-prd-rh01.pg1f.p1/product/ReleasePlanAdmission/ose-osc/osc-stage.yaml#L19

dbkreling commented on Dockerfile:
```diff
@@ -41,7 +41,7 @@ RUN getent group nonroot || groupadd -o -g 499 nonroot
 # Red Hat labels
 LABEL name="openshift-sandboxed-containers/osc-rhel9-operator" \
 cpe="cpe:/a:redhat:confidential_compute_attestation:1.10::el9" \
```
Same here, should we bump the cpe label here to 1.11?

littlejawa replied:
same answer :-)

dbkreling commented on scripts/bump-osc-version.sh:
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
```
Should we add a snippet with the usage of the script here, or else point the usage instructions in the development.md file to connect the two files?

dbkreling reviewed (commented):
Dropping some questions I was curious about if they should be changed now, later or ever.

dbkreling commented:
Oh, also, we have a reference here that was not updated. 
[config/peerpods/podvm/bootc/Containerfile.rhel#L2 ](https://github.com/gkurz/sandboxed-containers-operator/blob/11d0b7744b146db89a12705f5a8d125293b83c5d/config/peerpods/podvm/bootc/Containerfile.rhel#L2)
Is it relevant?

gkurz committed:
Drop patch level in version labels

It might happen that an image doesn't require actual changes when
bumping the patch level of OSC, but we end up generating a new
image with same content anyway because of the label change.

Drop the patch level from the version label to avoid that.

Signed-off-by: Greg Kurz <groug@kaod.org>

gkurz commented:
New version drops patch level in version labels, adds usage to the script and a new OSC_RELEASE tag for the bootc container file.

littlejawa reviewed (approved):
/lgtm
Thanks @gkurz !
